src/transformStateWithClones.js

```diff
@@ -1,13 +1,34 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const masiv = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const add in actions[i].extraData) {
+        newState[add] = actions[i].extraData[add];
+      }
+      masiv.push({ ...newState });
```

Comment:
we can move this out of the conditions and push only one time at the end of the loop